pkg/ha/source.go

```diff
@@ -28,6 +28,12 @@ import (
 // PutSourceCfg puts the config of the upstream source into etcd.
 // k/v: sourceID -> source config.
 func PutSourceCfg(cli *clientv3.Client, cfg *config.SourceConfig) (int64, error) {
+	if cfg.From.Security != nil {
+		// NOTE also put tls config data into etcd.
+		if err := cfg.From.Security.LoadTLSContent(); err != nil {
```

Comment:
this means the leader master should be accessible to certificates 🤔 I think it's not practical

Reply:
I can read the contents of the certificate into memory and transfer it to dm-master via rpc as soon as the user makes a request to create a source via `dmctl`

Reply:
fixed in  https://github.com/pingcap/dm/pull/1910/commits/e8499a0e8f887313b8930860bea179820c313839